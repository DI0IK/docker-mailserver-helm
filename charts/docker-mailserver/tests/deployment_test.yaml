suite: deployment tests
templates:
  - deployment.yaml
chart:
  version: 0.1.0
tests:
  - it: image and tag should end up in deployment
    set:
      image.name: bananas
      image.tag: in-pajamas
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -docker-mailserver$
      - equal:
          path: spec.template.spec.containers[0].image
          value: bananas:in-pajamas

  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: { }