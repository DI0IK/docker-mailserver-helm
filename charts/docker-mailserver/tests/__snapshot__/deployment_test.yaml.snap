manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/name: RELEASE-NAME-docker-mailserver
        chart: docker-mailserver-0.1.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-docker-mailserver
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/name: RELEASE-NAME-docker-mailserver
          release: RELEASE-NAME
      strategy:
        type: Recreate
      template:
        metadata:
          labels:
            app.kubernetes.io/name: RELEASE-NAME-docker-mailserver
            release: RELEASE-NAME
        spec:
          containers:
            - env:
                - name: ACCOUNT_PROVISIONER
                  value: null
                - name: AMAVIS_LOGLEVEL
                  value: "0"
                - name: CLAMAV_MESSAGE_SIZE_LIMIT
                  value: null
                - name: DEFAULT_RELAY_HOST
                  value: null
                - name: DMS_VMAIL_GID
                  value: null
                - name: DMS_VMAIL_UID
                  value: null
                - name: DOVECOT_AUTH_BIND
                  value: null
                - name: DOVECOT_INET_PROTOCOLS
                  value: all
                - name: DOVECOT_MAILBOX_FORMAT
                  value: maildir
                - name: DOVECOT_PASS_FILTER
                  value: null
                - name: DOVECOT_TLS
                  value: null
                - name: DOVECOT_USER_FILTER
                  value: null
                - name: ENABLE_AMAVIS
                  value: "0"
                - name: ENABLE_CLAMAV
                  value: "0"
                - name: ENABLE_DNSBL
                  value: "0"
                - name: ENABLE_FAIL2BAN
                  value: "0"
                - name: ENABLE_FETCHMAIL
                  value: "0"
                - name: ENABLE_GETMAIL
                  value: "0"
                - name: ENABLE_IMAP
                  value: "1"
                - name: ENABLE_MANAGESIEVE
                  value: null
                - name: ENABLE_OPENDKIM
                  value: "0"
                - name: ENABLE_OPENDMARC
                  value: "0"
                - name: ENABLE_POLICYD_SPF
                  value: "0"
                - name: ENABLE_POP3
                  value: null
                - name: ENABLE_POSTGREY
                  value: "0"
                - name: ENABLE_QUOTAS
                  value: "1"
                - name: ENABLE_RSPAMD
                  value: "1"
                - name: ENABLE_RSPAMD_REDIS
                  value: "1"
                - name: ENABLE_SASLAUTHD
                  value: "0"
                - name: ENABLE_SPAMASSASSIN
                  value: "0"
                - name: ENABLE_SPAMASSASSIN_KAM
                  value: "0"
                - name: ENABLE_SRS
                  value: "0"
                - name: ENABLE_UPDATE_CHECK
                  value: "1"
                - name: FAIL2BAN_BLOCKTYPE
                  value: drop
                - name: FETCHMAIL_PARALLEL
                  value: "0"
                - name: FETCHMAIL_POLL
                  value: "300"
                - name: GETMAIL_POLL
                  value: "5"
                - name: LDAP_BIND_DN
                  value: null
                - name: LDAP_BIND_PW
                  value: null
                - name: LDAP_QUERY_FILTER_ALIAS
                  value: null
                - name: LDAP_QUERY_FILTER_DOMAIN
                  value: null
                - name: LDAP_QUERY_FILTER_GROUP
                  value: null
                - name: LDAP_QUERY_FILTER_USER
                  value: null
                - name: LDAP_SEARCH_BASE
                  value: null
                - name: LDAP_SERVER_HOST
                  value: null
                - name: LDAP_START_TLS
                  value: null
                - name: LOGROTATE_INTERVAL
                  value: weekly
                - name: LOGWATCH_INTERVAL
                  value: null
                - name: LOGWATCH_RECIPIENT
                  value: null
                - name: LOGWATCH_SENDER
                  value: null
                - name: LOG_LEVEL
                  value: info
                - name: MARK_SPAM_AS_READ
                  value: "0"
                - name: MOVE_SPAM_TO_JUNK
                  value: "1"
                - name: NETWORK_INTERFACE
                  value: null
                - name: ONE_DIR
                  value: "1"
                - name: OVERRIDE_HOSTNAME
                  value: mail.example.com
                - name: PERMIT_DOCKER
                  value: none
                - name: PFLOGSUMM_RECIPIENT
                  value: null
                - name: PFLOGSUMM_SENDER
                  value: null
                - name: PFLOGSUMM_TRIGGER
                  value: null
                - name: POSTFIX_DAGENT
                  value: null
                - name: POSTFIX_INET_PROTOCOLS
                  value: all
                - name: POSTFIX_MAILBOX_SIZE_LIMIT
                  value: null
                - name: POSTFIX_MESSAGE_SIZE_LIMIT
                  value: null
                - name: POSTFIX_REJECT_UNKNOWN_CLIENT_HOSTNAME
                  value: "0"
                - name: POSTGREY_AUTO_WHITELIST_CLIENTS
                  value: "5"
                - name: POSTGREY_DELAY
                  value: "300"
                - name: POSTGREY_MAX_AGE
                  value: "35"
                - name: POSTGREY_TEXT
                  value: Delayed by Postgrey
                - name: POSTMASTER_ADDRESS
                  value: null
                - name: POSTSCREEN_ACTION
                  value: enforce
                - name: RELAY_HOST
                  value: null
                - name: RELAY_PASSWORD
                  value: null
                - name: RELAY_PORT
                  value: "25"
                - name: RELAY_USER
                  value: null
                - name: REPORT_RECIPIENT
                  value: null
                - name: REPORT_SENDER
                  value: null
                - name: RSPAMD_CHECK_AUTHENTICATED
                  value: "0"
                - name: RSPAMD_GREYLISTING
                  value: "0"
                - name: RSPAMD_HFILTER
                  value: "1"
                - name: RSPAMD_HFILTER_HOSTNAME_UNKNOWN_SCORE
                  value: "6"
                - name: RSPAMD_LEARN
                  value: "0"
                - name: SASLAUTHD_LDAP_AUTH_METHOD
                  value: null
                - name: SASLAUTHD_LDAP_BIND_DN
                  value: null
                - name: SASLAUTHD_LDAP_FILTER
                  value: null
                - name: SASLAUTHD_LDAP_MECH
                  value: null
                - name: SASLAUTHD_LDAP_PASSWORD
                  value: null
                - name: SASLAUTHD_LDAP_PASSWORD_ATTR
                  value: null
                - name: SASLAUTHD_LDAP_SEARCH_BASE
                  value: null
                - name: SASLAUTHD_LDAP_SERVER
                  value: null
                - name: SASLAUTHD_LDAP_START_TLS
                  value: null
                - name: SASLAUTHD_LDAP_TLS_CACERT_DIR
                  value: null
                - name: SASLAUTHD_LDAP_TLS_CACERT_FILE
                  value: null
                - name: SASLAUTHD_LDAP_TLS_CHECK_PEER
                  value: null
                - name: SASLAUTHD_MECHANISMS
                  value: null
                - name: SASLAUTHD_MECH_OPTIONS
                  value: null
                - name: SA_KILL
                  value: "10"
                - name: SA_SPAM_SUBJECT
                  value: '***SPAM*** '
                - name: SA_TAG
                  value: "2"
                - name: SA_TAG2
                  value: "6.31"
                - name: SMTP_ONLY
                  value: null
                - name: SPAMASSASSIN_SPAM_TO_INBOX
                  value: "1"
                - name: SPOOF_PROTECTION
                  value: null
                - name: SRS_EXCLUDE_DOMAINS
                  value: null
                - name: SRS_SECRET
                  value: null
                - name: SRS_SENDER_CLASSES
                  value: envelope_sender
                - name: SSL_ALT_CERT_PATH
                  value: null
                - name: SSL_ALT_KEY_PATH
                  value: null
                - name: SUPERVISOR_LOGLEVEL
                  value: null
                - name: TLS_LEVEL
                  value: null
                - name: TZ
                  value: null
                - name: UPDATE_CHECK_INTERVAL
                  value: 1d
                - name: VIRUSMAILS_DELETE_DELAY
                  value: null
              image: mailserver/docker-mailserver:13.3.1
              imagePullPolicy: IfNotPresent
              livenessProbe:
                exec:
                  command:
                    - /bin/bash
                    - -c
                    - supervisorctl status | grep -E "amavis|clamav|cron|dovecot|mailserver|opendkim|opendmarc|postfix|rsyslog" | grep RUNNING
                failureThreshold: 3
                initialDelaySeconds: 10
                timeoutSeconds: 5
              name: docker-mailserver
              ports:
                - containerPort: 25
                  name: smtp
                - containerPort: 465
                  name: submissions
                - containerPort: 587
                  name: submission
                - containerPort: 10465
                  name: subs-proxy
                - containerPort: 10587
                  name: sub-proxy
                - containerPort: 143
                  name: imap
                - containerPort: 993
                  name: imaps
                - containerPort: 10143
                  name: imap-proxy
                - containerPort: 10993
                  name: imaps-proxy
                - containerPort: 11334
                  name: rspamd
              readinessProbe:
                exec:
                  command:
                    - /bin/bash
                    - -c
                    - supervisorctl status | grep -E "mailserver|postfix" | grep RUNNING
                failureThreshold: 3
                initialDelaySeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: "2"
                  ephemeral-storage: 500Mi
                  memory: 2048Mi
                requests:
                  cpu: "1"
                  ephemeral-storage: 100Mi
                  memory: 1536Mi
              securityContext:
                privileged: false
                readOnlyRootFilesystem: false
              volumeMounts:
                - mountPath: /tmp/docker-mailserver/dovecot.cf
                  name: dovecot-cf
                  subPath: dovecot.cf
                - mountPath: /etc/dovecot/conf.d/10-plugin.conf
                  name: fts-xapian-plugin-conf
                  subPath: fts-xapian-plugin.conf
                - mountPath: /tmp/docker-mailserver/user-patches.sh
                  name: user-patches-sh
                  subPath: user-patches.sh
                - mountPath: /tmp/docker-mailserver
                  name: mail-config
                - mountPath: /var/mail
                  name: mail-data
                - mountPath: /var/log/mail
                  name: mail-log
                - mountPath: /var/mail-state
                  name: mail-state
          nodeSelector: {}
          priorityClassName: null
          restartPolicy: Always
          runtimeClassName: null
          securityContext: null
          serviceAccountName: RELEASE-NAME-docker-mailserver
          volumes:
            - configMap:
                name: dovecot-cf
              name: dovecot-cf
            - configMap:
                name: fts-xapian-plugin-conf
              name: fts-xapian-plugin-conf
            - configMap:
                name: user-patches-sh
              name: user-patches-sh
            - name: mail-config
              persistentVolumeClaim:
                claimName: RELEASE-NAME-docker-mailserver-mail-config
            - name: mail-data
              persistentVolumeClaim:
                claimName: RELEASE-NAME-docker-mailserver-mail-data
            - name: mail-log
              persistentVolumeClaim:
                claimName: RELEASE-NAME-docker-mailserver-mail-log
            - name: mail-state
              persistentVolumeClaim:
                claimName: RELEASE-NAME-docker-mailserver-mail-state
